src/main/kotlin/com/zepben/evolve/cim/iec61970/base/wires/SynchronousMachine.kt

```diff
@@ -0,0 +1,130 @@
+ * @property earthing Indicates whether the generator is earthed. Used for short circuit data exchange according to IEC 60909.
+ * @property earthingStarPointR Generator star point earthing resistance in Ohms (Re). Used for short circuit data exchange according to IEC 60909.
+ * @property earthingStarPointX Generator star point earthing reactance in Ohms (Xe). Used for short circuit data exchange according to IEC 60909.
+ * @property ikk Steady-state short-circuit current (in A for the profile) of generator with compound excitation during 3-phase short circuit.
+ *               - Ikk=0: Generator with no compound excitation. - Ikk<>0: Generator with compound excitation.
+ *               Ikk is used to calculate the minimum steady-state short-circuit current for generators with compound excitation.
+ *               (4.6.1.2 in IEC 60909-0:2001). Used only for single fed short circuit on a generator. (4.3.4.2. in IEC 60909-0:2001).
+ * @property maxQ Maximum reactive power limit in VAr. This is the maximum (nameplate) limit for the unit.
+ * @property maxU Maximum voltage limit for the unit in volts.
+ * @property minQ Minimum reactive power limit for the unit in VAr.
+ * @property minU Minimum voltage limit for the unit in volts.
+ * @property mu Factor to calculate the breaking current (Section 4.5.2.1 in IEC 60909-0).
+ *              Used only for single fed short circuit on a generator (Section 4.3.4.2. in IEC 60909-0).
+ * @property r Equivalent resistance (RG) of generator as a percentage. RG is considered for the calculation of all currents,
+ *             except for the calculation of the peak current ip. Used for short circuit data exchange according to IEC 60909.
+ * @property r0 Zero sequence resistance of the synchronous machine as a percentage.
+ * @property r2 Negative sequence resistance as a percentage.
+ * @property satDirectSubtransX Direct-axis subtransient reactance saturated as a percentage, also known as Xd"sat.
+ * @property satDirectSyncX Direct-axes saturated synchronous reactance (xdsat); reciprocal of short-circuit ration, as a percentage.
+ *                          Used for short circuit data exchange, only for single fed short circuit on a generator. (4.3.4.2. in IEC 60909-0:2001).
+ * @property satDirectTransX Saturated Direct-axis transient reactance as a percentage.
+ *                           The attribute is primarily used for short circuit calculations according to ANSI.
+ * @property x0 Zero sequence reactance of the synchronous machine as a percentage.
+ * @property x2 Negative sequence reactance as a percentage.
+ * @property type Modes that this synchronous machine can operate in.
+ * @property operatingMode Current mode of operation.
+ */
+class SynchronousMachine @JvmOverloads constructor(mRID: String = "") : RotatingMachine(mRID) {
+
+    private var _reactiveCapabilityCurve: MutableList<ReactiveCapabilityCurve>? = null
```

Comment:
todo: `_reactiveCapabilityCurve` should be pluralised as it is a list of curves.

Reply:
changed